tutorials/single_node_tutorial/single_gpu_tutorial.ipynb

```diff
@@ -0,0 +1,3806 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "62336143",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.cluster.close()\n",
+    "client.shutdown()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d6fb16b1",
+   "metadata": {},
+   "source": [
+    "## 4.Exact Dedplication\n",
+    "\n",
+    "In exact deduplication, the document text is hashed into unique string using certain hashing algorithm, such as 'md5'. The documents with exact hashed values are having identical text. We will output the `ID` of duplicated documents for removal later. The function used is `ExactDuplicates()`. Arguments for this function include:\n",
+    "- `id_field`: Key in input file for identifying document ID\n",
+    "- `text_field`: Key in input file which contains document text.\n",
+    "- `hash_method`: Hashing algorithm used. Default is `md5`\n",
+    "- `cache_dir`: If specified, the duplicated document IDs will be output to the `cache_dir`. Otherwise, the IDs will not be saved\n",
+    "\n",
+    "Also, we are going to use GPU dask cluster to accelerate computation for deduplication (both exact and fuzzy)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "044f7eee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nemo_curator.modules import ExactDuplicates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e5da88e",
+   "metadata": {},
+   "source": [
+    "Start a GPU based Dask cluster. Since GPU based Dask cluster involves setting several arguments, we will use the `get_client()` wrapper function to quickly set up. Please make sure the `device` in `args` is `gpu`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "e4d6920d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Namespace(scheduler_address=None, scheduler_file=None, n_workers=20, threads_per_worker=1, rmm_pool_size=None, protocol='tcp', nvlink_only=False, files_per_partition=2, num_files=None, device='gpu', set_torch_to_use_rmm=False)"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sys.argv=['','--device','gpu']\n",
+    "parser = argparse.ArgumentParser()\n",
+    "args = attach_args(parser).parse_args()\n",
+    "args.set_torch_to_use_rmm = False\n",
+    "args"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "717b6cef",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of dask worker:1\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'tcp://127.0.0.1:42505': None}"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "client = get_client(args, args.device)\n",
```

Comment:
Ah we just merged in a change that changes the function signature of this to no longer require an argparse object. It should be easier to use, but it does mean that you need to update it here. Ping me again when this is changed and I'll merge it in ASAP.

Reply:
Thanks for noting. I have updated accordingly.